zap/src/main/java/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -1300,6 +1302,9 @@ protected boolean isCustomPage(HttpMessage msg, CustomPage.Type cpType) {
         if (getContext() != null) {
             return getContext().isCustomPageWithFallback(msg, cpType);
         }
+        if (CustomPage.Type.NOTFOUND_404.equals(cpType)) {
+            return !getAnalyser().isFileExist(msg);
+        }
```

Comment:
I actually originally thought hooking analyzer was going to be more involved, but with our most recent changes I believe this is all that's needed.

Also feel free to correct me if the logic is wrong but this is how I was considering it: `isCustomPage` checks if this matches a 404 definition and returns true which is the inverse of `isFileExist` (it returns false in 404 cases, but as far as custom pages are concerned if it is a 404 then we want true).